app/javascript/mastodon/components/router.tsx

```diff
@@ -20,39 +24,45 @@ const browserHistory = createBrowserHistory<
 const originalPush = browserHistory.push.bind(browserHistory);
 const originalReplace = browserHistory.replace.bind(browserHistory);
 
-function extractRealPath(path: HistoryPath) {
-  if (typeof path === 'string') return path;
-  else return path.pathname;
+function normalizePath(
+  path: HistoryPath,
+  state?: MastodonLocationState,
+): LocationDescriptorObject<MastodonLocationState> {
+  path =
+    typeof path === 'string' ? { pathname: path, state: state } : { ...path };
```

Comment:
If there is an object path + a `state`, then we ignore the `state`?

Should we merge it instead?

Reply:
The state is already part of the `path` object, and upstream's `history` module ignores the second argument in this case (with a warning). We pass the second argument along, so we do get the same warning in case of misuse.

Reply:
Wait, we will get the warning even in case of proper use because we always return a path object :thinking:

Reply:
This should be fixed, now.